### DISCUSSION TITLE
Is there a way to toggle `colors.webpage.darkmode.enabled` ?

### DISCUSSION THREAD

**Discussion by danielfalbo** [qutebrowser/qutebrowser#5767] [Q&A] [answered]
I currently have
```
# Pseudo dark mode
config.set("colors.webpage.darkmode.enabled", True)
config.set("colors.webpage.darkmode.policy.images", "smart")
```
in my config.py and I would like to make some key toggle it but just doing `set` or `config-cycle` is not enough, this requires restart, but when I restart qutebrowser it reads the config.py file and uses the setting from the file.

**Comment by The-Compiler** [comment, marked as answer]
There's no way around the restart unfortunately - this would need changes in Chromium and/or QtWebEngine.

You could [load your `autoconfig.yml`](https://github.com/qutebrowser/qutebrowser/blob/master/doc/help/configuring.asciidoc#loading-autoconfigyml) after the `config.set` calls so that the settings in there take priority.

You might also be able to do something like `os.environ['QUTE_DARKMODE_ENABLED'] = 1` and `if 'QUTE_DARKMODE_ENABLED' not in os.environ: ...` if using `:restart`, though I haven't tested and I'm not entirely sure if `:restart` will preserve environment variables.

**Comment by danielfalbo** [comment]
Works :) Thank you very much

**Comment by edrex** [comment]
I'm researching options for globally toggling dark mode on Linux (desktop independent). I am thinking a service with a dbus interface to toggle, and plugins for different applications/toolkits.

For qutebrowser some method for dynamically toggling mode is needed. I think a good user story would be to document a key binding that switches it on.

Chromium's dark mode [is enabled through shell setting on Win/Mac](https://support.google.com/chrome/answer/9275525?co=GENIE.Platform%3DDesktop&hl=en). I'm not sure if this is implemented via a Chrome restart (TODO(edrex):test with Mavericks Chromium). Whatever Chromium is doing, I'd like to implement similar in qutebrowser. If Chromium is able to toggle without a restart, I'd like to get a patch into QTWebEngine to plumb that through. If a restart is required, well, that is unfortunate for me since I use tabs-as-windows and restarting flattens my window placement but c'est la vie.

PS: Chromium's dark mode is fantastic. I've been running it as the default for awhile and have very few issues.

**Reply by edrex** [reply]
Florian's detailed research here: https://bugreports.qt.io/browse/QTBUG-84484?jql=project%20%3D%20QTBUG%20AND%20component%20%3D%20WebEngine%20and%20text%20~%20%22dark%20mode%22

- [ ] edrex come up with a bind that toggles the setting and restarts, for the moment

**Reply by The-Compiler** [reply]
Also see https://github.com/qutebrowser/qutebrowser/issues/5542#issuecomment-782040210 for a hack to disable it dynamically. You can probably do the same via `:jseval` or so as well. I haven't really done any sophisticated testing with it, though.

**Reply by edrex** [reply]
Interesting.. 
I'm most interested in a global toggle. I could try the userscript route, bt seems a bit brittle and I'd prob rather wait for things to shake out in chromium. Thanks for the experimentation and sharing your results!
